llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -426,6 +429,21 @@ static unsigned getMinSchedulableCycle(SchedBoundary &Zone) {
   return MinSchedulableCycle;
 }
 
+// This function returns false when the available queue is empty and there is a
+// single instruction in the pending queue that has a delay slot. Continuing
+// with a top-down approach in this scenario would lead to an infinite loop,
+// since instructions with delay slots are never available for the top zones.
```

Comment:
nit: I think the last observation is more important. In fact, progress is blocked if no instruction in the pending queue can become available in top down. The fact that currently only delayslot instructions apply and that we can only have one delay slot instruction in a region is a coincidence

Reply:
I'm wondering if we should remove the instruction from the pending queue altogether. Basically never have it in the queues of the Top zone.